connectors/tests/test_sync.py

```diff
@@ -486,7 +440,8 @@ def bulk_call(url, **kw):
         return CallbackResult(status=200, payload={"items": []})
 
     await set_server_responses(mock_responses, FAKE_CONFIG_TS, bulk_call=bulk_call)
-    service = create_service(CONFIG_FILE, sync_now=True, one_sync=True)
+    service = create_service(CONFIG_FILE)
+    asyncio.get_event_loop().call_later(10, service.stop)
```

Comment:
This test will run for 10 seconds now which will make the tests run much slower, is it possible to make it run fast?

In Ruby I was mocking `sleep` and `loop`, for example - `sleep` did nothing, `loop` ran iteration only once, do you think it can be used here?

Reply:
I don't know whether I can mock `while self.running`.

`service.stop` is used to set `self.running` to `false`.

Reply:
cc @tarekziade - how can we make this test fast?

Reply:
yeah we can:

```
task = asyncio.create_task(service.run())
task.add_done_callback(service.stop)
await task
```

Reply:
same pattern but we can switch the flag after 200ms, the sync will already be running, and the loop will exit after it's done, in the next iteration

Reply:
This way, we assume that in 200 ms, the service has done pre-flight check, and has done at least one loop. 200 ms may not be enough once we have decoupled producer and consumer so our service is more async.

Actually, this is no different from
https://github.com/elastic/connectors-python/blob/1faa0098c513f70d9b396bcad0a22eec7d53ee61/connectors/tests/test_sync.py#L449

where we can just reduce from 10 to 0.2

Reply:
> has done at least one loop
has entered the loop, next check is when you do the loop again even if the first iteration took longer than 200ms

Reply:
Exactly. my point is it's the same as

```python
asyncio.get_event_loop().call_later(0.2, service.stop) 
```